File: 7kyu 1/alphabetWar.js
function alphabetWar(fight) {
   let left = 0;
   let right = 0;
   for (let i = 0; i < fight.length; i++) {
      if (fight[i] == 'w' || fight[i] == 'p' || fight[i] == 'b' || fight[i] == 's') {
         left += fight[i] == 'w' ? 4 : fight[i] == 'p' ? 3 : fight[i] == 'b' ? 2 : fight[i] == 's' ? 1 : 0;
      } else {
         right += fight[i] == 'm' ? 4 : fight[i] == 'q' ? 3 : fight[i] == 'd' ? 2 : fight[i] == 'z' ? 1 : 0;
      }
   }
   return left > right ? 'Left side wins!' : left < right ? 'Right side wins!' : "Let's fight again!";
}


function alphabetWar(fight) {
   let map = {
      w: -4,
      p: -3,
      b: -2,
      s: -1,
      m: 4,
      q: 3,
      d: 2,
      z: 1
   };
   let result = fight.split('').reduce((a, b) => a + (map[b] || 0), 0);
   return result ? (result < 0 ? "Left" : "Right") + " side wins!" : "Let's fight again!";
}


function alphabetWar(fight) {
   thisIsWar = {
      'w': 4,
      'p': 3,
      'b': 2,
      's': 1,
      'm': -4,
      'q': -3,
      'd': -2,
      'z': -1
   };
   result = 0;
   fight.split("").map(function (l) {
      result += thisIsWar[l] !== undefined ? thisIsWar[l] : 0
   });
   return result < 0 ? "Right side wins!" : result > 0 ? "Left side wins!" : "Let's fight again!"
}


function alphabetWar(fight) {
   let score = 0;
   for (const letter of fight) {
      switch (letter) {
         case 'w':
            score -= 4;
            break;
         case 'p':
            score -= 3;
            break;
         case 'b':
            score -= 2;
            break;
         case 's':
            score -= 1;
            break;
         case 'm':
            score += 4;
            break;
         case 'q':
            score += 3;
            break;
         case 'd':
            score += 2;
            break;
         case 'z':
            score += 1;
            break;
      }
   }
   return score < 0 ? "Left side wins!" : score > 0 ? "Right side wins!" : "Let's fight again!";
}


console.log(alphabetWar("z"), "Right side wins!");
console.log(alphabetWar("zdqmwpbs"), "Let's fight again!");
console.log(alphabetWar("zzzzs"), "Right side wins!");
console.log(alphabetWar("wwwwww"), "Left side wins!");

File: Python/7py 1/alphabet_war.py
"""
|--------------------------------------------------------------------------
| Alphabet war
|--------------------------------------------------------------------------
|
"""

def alphabet_war(fight):
   score = 0
   for i in range(len(fight)):
      if fight[i] == 'w':
         score -= 4
      elif fight[i] == 'p':
         score -= 3
      elif fight[i] == 'b':
         score -= 2
      elif fight[i] == 's':
         score -= 1
      elif fight[i] == 'm':
         score += 4
      elif fight[i] == 'q':
         score += 3
      elif fight[i] == 'd':
         score += 2
      elif fight[i] == 'z':
         score += 1
   return "Left side wins!" if score < 0 else "Right side wins!" if score > 0 else "Let's fight again!"


def alphabet_war(fight):
   result = sum("zdqm".find(c) - "sbpw".find(c) for c in fight)
   return "{} side wins!".format("Left" if result < 0 else "Right") if result else "Let's fight again!"


def alphabet_war(fight):
   d = {'w':4,'p':3,'b':2,'s':1,
      'm':-4,'q':-3,'d':-2,'z':-1}
   r = sum(d[c] for c in fight if c in d)
   return {r==0:"Let's fight again!",
         r>0:"Left side wins!",
         r<0:"Right side wins!"
         }[True]


POWERS  = {c:i for i,c in enumerate('wpbs zdqm',-4)}
def alphabet_war(fight):
   stroke = sum(POWERS[c] for c in fight if c != ' ' and c in POWERS)
   return ['Left side wins!', "Let's fight again!", 'Right side wins!'][ (stroke >= 0) + (stroke > 0) ]


print(alphabet_war("z"), "Right side wins!")
print(alphabet_war("zdqmwpbs"), "Let's fight again!")
print(alphabet_war("wq"), "Left side wins!")
print(alphabet_war("zzzzs"), "Right side wins!")
print(alphabet_war("wwwwww"), "Left side wins!")